addons/website_event_track_quiz/controllers/main.py

```diff
@@ -0,0 +1,144 @@
+import json
+
+from odoo import http
+from odoo.addons.website_event_track_session.controllers.session import WebsiteEventSessionController
+from odoo.exceptions import AccessError, UserError
+from odoo.http import request
+
+
+class WebsiteEventTrackQuiz(WebsiteEventSessionController):
+    def _fetch_track(self, track_id):
+        track = request.env['event.track'].browse(int(track_id)).exists()
+        if not track:
+            return {'error': 'track_wrong'}
+        try:
+            track.check_access_rights('read')
+            track.check_access_rule('read')
+        except AccessError:
+            return {'error': 'track_access'}
+        return {'track': track}
+
+    def _event_track_get_values(self, event, track, **options):
+        values = super(WebsiteEventTrackQuiz, self)._event_track_get_values(event, track)
+        if 'quiz' in options:
+            values.update({
+                'show_quiz': True,
+                'visitor': self._get_event_track_visitor_info(track)
+            })
+        return values
+
+    def _get_event_track_visitor_info(self, track):
+        track_visitor = request.env['event.track.visitor'].sudo().search([('track_id', '=', track.id), ('partner_id', '=', request.env.user.partner_id.id)])
+        return {
+            'quiz_completed': track_visitor.quiz_completed if track_visitor else 0,
+            'points_gained': track_visitor.points_gained if track_visitor else 0
+        }
+
+    def _get_event_track_quiz_data(self, track):
+        track_visitor = self._get_event_track_visitor(track)
+        values = {
+            'slide_questions': [{
+                'id': question.id,
+                'question': question.question,
+                'answer_ids': [{
+                    'id': answer.id,
+                    'text_value': answer.text_value,
+                    'is_correct': answer.is_correct if track_visitor.quiz_completed or request.website.is_publisher() else None,
+                    'comment': answer.comment if request.website.is_publisher else None
+                } for answer in question.sudo().answer_ids],
+            } for question in track.quiz_id.question_ids]
+        }
+        if 'slide_answer_quiz' in request.session:
+            slide_answer_quiz = json.loads(request.session['track_answer_quiz'])
+            if str(track.id) in slide_answer_quiz:
+                values['session_answers'] = slide_answer_quiz[str(track.id)]
+        values.update(self._get_event_track_visitor(track))
+        return values
+
+    def _get_event_track_visitor(self, track):
```

Comment:
Probably to merge with the up-mentionned _find_track_visitor method, that could receive a force_create parameter (either find, either create a new relationship and returns it).